pandas/tseries/index.py

```diff
@@ -1507,6 +1508,16 @@ def date(self):
         """
         return self._maybe_mask_results(_algos.arrmap_object(self.asobject.values, lambda x: x.date()))
 
+    @property
+    def weekday_name(self):
+        """
+        Returns numpy array of weekday name string. The name of day in a week (ex: Friday)
+
+        .. versionadded:: 0.18.0
+        """
+        weekdays = [calendar.day_name[i].lower() for i in range(7)]
+        return self._maybe_mask_results(_algos.arrmap_object(self.asobject.values, lambda x: weekdays[x.weekday()]))
+
```

Comment:
I'm totally noob about `tslib.pyx` because it is a Cython code that interacting with numpy, CPython datetime. I wanna give my best shot to update the implementation on the `tslib.pyx`. And then, acces it via `_field_accessor` in `tseries/index.py` with additional if branching checking if it is requesting a `weekday_name`. And we also can add like `month_name` perhaps? What do you think @jreback ?